### PR TITLE
Add an ability to create and edit namespaced secrets by workspace user

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
To be able to create edit and delete namespaced secretes workspace user needs special permissions. This PR adds a kubernetes role for secrets, and binds it to a workspace user.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19837

### How to test this PR?

1. Start a workspace and open a terminal from ide container
2. Send an HTTP request to kubernetes API to create a secret: `curl -X POST <kubernetes API url>/api/v1/namespaces/<namespace name>/secrets --header "Content-Type: application/json" -d '{ "apiVersion": "v1", "kind": "Secret", "metadata": { "name" : "new-secret" } }' `

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by sleshchenko — COMMENTED** [review]
pay attention that it won't work for existing namespaces, only newly created due https://github.com/eclipse/che/issues/19697

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java [hunk 1]
```diff
@@ -133,6 +134,15 @@ private void createImplicitRolesWithBindings(Client k8sClient) {
         singletonList("metrics.k8s.io"),
         Arrays.asList("list", "get", "watch"),
         serviceAccountName + "-metrics");
+
+    // secrets role
+    createRoleWithBinding(
```
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java:139]: note that it won't work for DevWorkspaces.
Please contribute https://github.com/devfile/devworkspace-operator/blob/main/controllers/workspace/provision/rbac.go#L43 or create an issue for us https://github.com/devfile/devworkspace-operator/issues/new

Comment (vinokurig) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java:139]: I've created an issue for that: https://github.com/devfile/devworkspace-operator/issues/483

**Comment by skabashnyuk** [comment]
Do we need documentation for that?

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java [hunk 2]
```diff
@@ -133,6 +134,15 @@ private void createImplicitRolesWithBindings(Client k8sClient) {
...
+        k8sClient,
+        SECRETS_ROLE_NAME,
+        Arrays.asList("secrets"),
+        singletonList(""),
+        Arrays.asList("list", "create", "delete"),
```
Comment (skabashnyuk) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java:144]: is there a way to test these roles?

**Comment by skabashnyuk** [comment]
LGTM to me in general. Let's see that other folks are thinking about that.

**Review by sparkoo — APPROVED** [review]
lgtm

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java (hunk 2 above)
Comment (vinokurig) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java:144]: I've added a testcase but all tests in this module are ignored. Should I create an issue for that?

**Comment by vinokurig** [comment]
> Do we need documentation for that?

I don't think so because it doesn't affect user experience. We might add some for the related issue.

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**